pkg/storage/replica.go

```diff
@@ -3301,6 +3308,12 @@ func (r *Replica) processRaftCommand(
 			"command %s proposed from replica %+v: %s",
 			raftCmd.BatchRequest, raftCmd.OriginReplica, err,
 		)
+		// TODO(spencer,andrei): This error is confusing if r.mu.state.Lease is
```

Comment:
We should probably return something different in that case. Or else update the code in `roachpb/errors.go` for `NotLeaseHolderError` to print something more sensible if the lease replica and replica are the same.